src/mesh/voxel_cloud.rs

```diff
@@ -0,0 +1,499 @@
+                        // is shifted -1, -1, -1)
+                        voxel_cloud
+                            .set_relative(Voxel::On, &(voxel_coords + Vector3::new(1, 1, 1)));
+                        // as well as all its neighbors
+                        voxel_cloud
+                            .set_relative(Voxel::On, &(voxel_coords + Vector3::new(0, 1, 1)));
+                        voxel_cloud
+                            .set_relative(Voxel::On, &(voxel_coords + Vector3::new(2, 1, 1)));
+                        voxel_cloud
+                            .set_relative(Voxel::On, &(voxel_coords + Vector3::new(1, 0, 1)));
+                        voxel_cloud
+                            .set_relative(Voxel::On, &(voxel_coords + Vector3::new(1, 2, 1)));
+                        voxel_cloud
+                            .set_relative(Voxel::On, &(voxel_coords + Vector3::new(1, 1, 0)));
+                        voxel_cloud
+                            .set_relative(Voxel::On, &(voxel_coords + Vector3::new(1, 1, 2)));
+                    }
+                }
+            }
+        }
+
+        voxel_cloud
+    }
+
+    /// Gets the index to the voxel map from voxel coordinates relative to the
+    /// voxel space block start.
+    ///
+    /// The function doesn't clamp the result because it may be used also to
+    /// check if the coordinates are out of bounds.
+    fn relative_three_dimensional_coordinate_to_one_dimensional(
+        &self,
+        relative_coords: &Point3<i32>,
+    ) -> i32 {
+        relative_coords.z * self.block_dimensions.x as i32 * self.block_dimensions.y as i32
+            + relative_coords.y * self.block_dimensions.x as i32
+            + relative_coords.x
+    }
+
+    /// Gets the index to the voxel map from absolute voxel coordinates
+    /// (relative to the voxel space origin).
+    ///
+    /// The function doesn't clamp the result because it may be used also to
+    /// check if the coordinates are out of bounds.
+    fn absolute_three_dimensional_coordinate_to_one_dimensional(
+        &self,
+        absolute_coords: &Point3<i32>,
+    ) -> i32 {
+        let relative_coords = absolute_coords - self.block_start.coords;
+        self.relative_three_dimensional_coordinate_to_one_dimensional(&relative_coords)
+    }
+
+    /// Calculates the voxel-space coordinates of a voxel containing the input
+    /// point.
+    fn cartesian_to_voxel_coords(&self, point: &Point3<f32>) -> Point3<i32> {
+        Point3::new(
+            (point.x / self.voxel_dimensions.x).round() as i32,
+            (point.y / self.voxel_dimensions.y).round() as i32,
+            (point.z / self.voxel_dimensions.z).round() as i32,
+        )
+    }
+
+    /// Calculates the center of a voxel in model-space coordinates from
+    /// absolute voxel coordinates (relative to the voxel space origin).
+    #[allow(dead_code)]
+    fn absolute_voxel_to_cartesian_coords(&self, absolute_coords: &Point3<i32>) -> Point3<f32> {
```

Comment:
Disagree! The `VoxelCloud` is a prominent data type and will soon be as important as the `Mesh` itself. Therefore whenever I'm implementing any functionality (in this case coordinate conversion), I'm doing all of it. Besides, this will be useful for volumetric boolean operations.